test/L2ArbitrumGovernorV2.t.sol

```diff
@@ -196,6 +206,96 @@ abstract contract Quorum is L2ArbitrumGovernorV2Test {
 
+abstract contract Propose is L2ArbitrumGovernorV2Test {
+  event ProposalCreated(
+    uint256 proposalId,
+    address proposer,
+    address[] targets,
+    uint256[] values,
+    string[] signatures,
+    bytes[] calldatas,
+    uint256 voteStart,
+    uint256 voteEnd,
+    string description
+  );
+
+  function testFuzz_Propose(uint256 _actorSeed) public {
+    // Proposal parameters
+    address[] memory targets = new address[](1);
+    uint256[] memory values = new uint256[](1);
+    bytes[] memory calldatas = new bytes[](1);
+    string[] memory signatures = new string[](1);
+    uint256 voteStart = vm.getBlockNumber() + governor.votingDelay();
+    uint256 voteEnd = voteStart + governor.votingPeriod();
+    string memory description = "Test";
+
+    uint256 proposalId = governor.hashProposal(targets, values, calldatas, keccak256(bytes(description)));
+    address _actor = _getMajorDelegate(_actorSeed);
+    vm.prank(_actor);
+    vm.expectEmit();
+    emit ProposalCreated(proposalId, _actor, targets, values, signatures, calldatas, voteStart, voteEnd, description);
+    governor.propose(targets, values, calldatas, description);
+
+    assertEq(uint256(governor.state(proposalId)), uint256(IGovernor.ProposalState.Pending));
+  }
+}
+
+abstract contract Cancel is L2ArbitrumGovernorV2Test {
+  event ProposalCanceled(uint256 proposalId);
+
+  error GovernorOnlyProposer(address proposer);
+  error ProposalNotPending(IGovernor.ProposalState state);
+
+  function testFuzz_CancelProposalWhenPending(uint256 _actorSeed) public virtual {
+    address[] memory targets = new address[](1);
+    uint256[] memory values = new uint256[](1);
+    bytes[] memory calldatas = new bytes[](1);
+    string memory description = "Test";
+
+    address _actor = _getMajorDelegate(_actorSeed);
+    vm.prank(_actor);
+    uint256 proposalId = governor.propose(targets, values, calldatas, description);
+
+    assertEq(uint256(governor.state(proposalId)), uint256(IGovernor.ProposalState.Pending));
+    vm.prank(address(_actor));
+    governor.cancel(targets, values, calldatas, keccak256(bytes(description)));
+    assertEq(uint256(governor.state(proposalId)), uint256(IGovernor.ProposalState.Canceled));
+  }
+
+  function testFuzz_RevertIf_NotProposer(uint256 _actorSeed, address _actor) public {
+    address _proposer = _getMajorDelegate(_actorSeed);
+    vm.assume(_actor != _proposer && _actor != PROXY_ADMIN_CONTRACT);
+    address[] memory targets = new address[](1);
+    uint256[] memory values = new uint256[](1);
+    bytes[] memory calldatas = new bytes[](1);
+    string memory description = "Test";
+
+    vm.prank(_proposer);
+    governor.propose(targets, values, calldatas, description);
+
+    vm.prank(address(_actor));
+    vm.expectRevert(abi.encodeWithSelector(GovernorOnlyProposer.selector, _actor));
+    governor.cancel(targets, values, calldatas, keccak256(bytes(description)));
+  }
+
+  function testFuzz_RevertIf_ProposalIsActive(uint256 _actorSeed) public {
```

Comment:
wonder if we can use `mockCall` to simulate all proposal states besides `Pending`

Reply:
oh good idea!